waspc/src/Wasp/Util/Json.hs

```diff
@@ -0,0 +1,16 @@
+module Wasp.Util.Json (parseJsonWithComments) where
+
+import Data.Aeson (FromJSON)
+import System.Exit (ExitCode (..))
+import qualified System.Process as P
+import Wasp.Util.Aeson (decodeFromString)
+
+-- | Uses Node.js to parse JSON with comments by treating it as a JavaScript object.
+parseJsonWithComments :: FromJSON a => String -> IO (Either String a)
+parseJsonWithComments jsonStr = do
+  let evalScript = "const v = " ++ jsonStr ++ ";console.log(JSON.stringify(v));"
```

Comment:
This is a trick we used to parse JSON with comment successfully. Aeson can't handle it. Other options included using the `tsc --showConfig` which returns something similar to the user's config, but it's still modified by `tsc` a bit (paths are absolute). I also considered implementing my own parser that would drop comments and just return a string we'll then parse - but the Node.js option worked so well and it's not that costly to invoke IMHO.

Reply:
Cool! Could you add some of this GH comment to the actual comment in the code? I think whoever will be reading this code will benefit from knowing more about why this was done (Aeson can't handle it) and why it was needed.